utilities/cli/holohub.py

```diff
@@ -916,13 +1057,19 @@ def handle_run(self, args: argparse.Namespace) -> None:
                     base_img=args.base_img,
                     img=args.img,
                     no_cache=args.no_cache,
-                    build_args=args.build_args,
+                    build_args=build_args.get("build_args"),
                 )
             language = holohub_cli_util.normalize_language(
                 container.project_metadata.get("metadata", {}).get("language", None)
             )
 
-            run_cmd = f"{self.script_name} run {args.project} --language {language} --local"
+            run_cmd = f"{self.script_name} run {args.project}"
+            if mode_name:
+                run_cmd += f" {mode_name}"
+            run_cmd += f" --language {language}"
+            # Only add --local if we're not in a nested container scenario
+            if not (mode_config and "run" in mode_config and "docker_args" in mode_config["run"]):
+                run_cmd += " --local"
```

Comment:
Not sure I follow, could you please clarify the "nested container scenario" here? Are we trying to support docker-in-docker?